### PR TITLE
Properly pass along actual referenced symbol in FindCallersAsync result.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Looking for right place to add tests.

### CODE REVIEW

File: src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs [hunk 1]
```diff
@@ -301,37 +301,38 @@ private static bool IsAccessible(SymbolAndProjectId symbolAndProjectId)
...
+            return results;
+
+            async Task AddReferencingSymbols(ReferencedSymbol reference, bool isDirect)
+            {
+                var result = await reference.Locations.FindReferencingSymbolsAsync(cancellationToken).ConfigureAwait(false);
+                foreach (var (callingSymbol, locations) in result)
                 {
-                    results ??= new List<SymbolCallerInfo>();
-                    results.Add(new SymbolCallerInfo(kvp.Key, symbol, kvp.Value, isDirect: false));
+                    results.Add(new SymbolCallerInfo(callingSymbol, reference.Definition, locations, isDirect));
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs]: this line is the primary change.  Previously, we would say that these locations were references to the `symbol` passed into `FindCallersAsync` (modulo getting the `original source definition`).  However, this wasn't really true as these were often references to something like an interface method (or a base/override).  Now we actually pass along the real symbol that was referenced.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @genlu

File: src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs (hunk 1 above)
Comment (mavasani) [reply src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs]: Can you share a before and after screenshot for some sample code?

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Hierarchy.cs]: @mavasani This actually only affects callers of the API.  All features inside Roslyn already only filter to the 'direct' references.  So this is only about what people see when they care about 'indirect references'.

**Comment by mavasani** [comment]
@CyrusNajmabadi Are you planning to add unit tests?

**Comment by CyrusNajmabadi** [comment]
@mavasani Yes, i think that would make sense.  I'm surprised we have a public API with 0 tests, but i suppose that sort of thing happens :)

**Pull request merged**